File: DonationSite.Endpoint.API/ClientApp/DonationSite/src/models/site.ts
import * as Interfaces from '../interfaces/site';

export class CreateSite implements Interfaces.CreateSite {
  public createdDateTime: Date = new Date();
  public name: string = '';
  public url: string = '';
  public siteID: number = 0;
}

export class UpdateSite implements Interfaces.UpdateSite {
  public name: string = '';
  public url: string = '';
  public siteID: number = 0;
}

File: DonationSite.Core.Entities/Site/Site.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DonationSite.Core.Entities.Site
{
    public class Site
    {
        public int SiteID { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string URL { get; set; }
        public DateTime CreatedDateTime { get; set; }
        //public ICollection<Donate.Donate> Donates { get; set; }

    }
}
